docs/Deprecated-functionality.rst

```diff
@@ -34,11 +34,43 @@ For authors of easyconfig files:
 
 For developers of easyblocks:
 
-*(nothing yet)*
+* ref:`depr_error_reporting`
 
 For EasyBuild framework developers:
 
-*(nothing yet)*
+* ref:`depr_error_reporting`
+
+.. _depr_error_reporting:
+
+Report errors by raising ``EasyBuildError`` rather than using ``error()``/``exception()`` log methods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Use of the** ``error`` **and** ``exception`` **log methods is deprecated.**
```

Comment:
They shouldn't be used at all anymore, since they will still raise an EasyBuildError for now, but will no longer in v3.0 when we remove the custom definitions.

I clarified the text a little bit to reflect that better.